torch/csrc/autograd/functions/init.cpp

```diff
@@ -62,27 +63,89 @@ struct NoCtor {
 };
 
 template<typename C, typename T>
-static void addClass(PyObject* module, PyTypeObject& type, const char* name)
+static void addClass(PyObject* module, PyTypeObject& type, const char* name,
+  PyGetSetDef* function_properties=NULL, PyMethodDef* function_methods=NULL)
 {
-  createForwardFunctionPyTypeObject<T>(type, name);
+  createForwardFunctionPyTypeObject<T>(type, name, function_properties, function_methods);
   Py_INCREF(&type);
   PyModule_AddObject(module, name, (PyObject*)&type);
   registerCppFunction(typeid(C), &type);
 }
 
+template<typename T, typename ValueT, typename ParamsT, ValueT ParamsT::*ptr,
+         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
+PyObject* getTupleAttr(PyObject* obj, void* _unused)
+{
+  HANDLE_TH_ERRORS
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& arr = (T*)(self->cdata.get())->*ptr;
```

Comment:
Can you please parenthesize this expression properly? It's unclear what happens first - the cast or pointer to member dereference?